src/geb/trans.lisp

```diff
@@ -189,3 +189,178 @@ should calculate
+                              (seqn:composition
+                               (seqn:remove-right (seqwidth lft))
+                               (seqn:parallel-seq (drop-width (butlast
+                                                               (cdr (seqwidth cpr))
+                                                               (- (length (seqwidth rt))
+                                                                  (length (seqwidth lft))))
+                                                              (seqwidth lft))
+                                                  (drop-nil (nthcdr (1+ (length (seqwidth lft)))
+                                                                    (seqwidth cpr))))))
+            ;; branch on the left counts the first entries of an occuring and drops the rest
+            ;; then injects it into the smaller bit sizes if necessary
+            ;; (max(a1, b1),,,.,(max (an, bn),..., max(0, bm))) -> (a1,...,an,0)
+            ;; -> (a1,....,an)
+            (composition (to-seqn (mcadr obj))
+                         (drop-width (cdr (seqwidth cpr)) (seqwidth rt)))))
+          ;; branch on the right is just inejcting
+          ;; (max(a1, b1),..., max(0, bn)) -> (b1, ...., bn) -> codom))
+          ((< (length (seqwidth rt))
+              (length (seqwidth lft)))
+           (branch-seq (composition (to-seqn (mcar obj))
+                                    (drop-width (cdr (seqwidth cpr)) (seqwidth lft)))
+                       (composition (to-seqn (mcadr obj))
+                                    (composition (remove-right (seqwidth rt))
+                                                 (parallel-seq (drop-width (butlast
+                                                                            (cdr (seqwidth cpr))
+                                                                            (- (length (seqwidth lft))
+                                                                               (length (seqwidth rt))))
+                                                                           (seqwidth rt))
+                                                               (drop-nil (nthcdr (1+ (length (seqwidth rt)))
+                                                                                 (seqwidth cpr))))))))
+          ((= (length (seqwidth rt))
+              (length (seqwidth lft)))
+           (branch-seq (composition (to-seqn (mcar obj))
+                                    (drop-width (cdr (seqwidth cpr)) (seqwidth lft)))
+                       (composition (to-seqn (mcadr obj))
+                                    (drop-width (cdr (seqwidth cpr)) (seqwidth rt))))))))
+
+(defmethod to-seqn ((obj project-left))
+  "Takes a list of an even length and cuts the right half"
+  (let ((sw (seqwidth (mcar obj))))
+    (composition (remove-right sw)
+                 (parallel-seq (id sw)
+                               (drop-nil (seqwidth (mcadr obj)))))))
```

Comment:
why are we ending up with nils anyways. Seems like sloppy coding? Or does it have a point?

Reply:
With the constructors we currently have for the SeqN category this is the canonical way to do this. We can have extra constructors just skipping entries on left-right instead of only doing this for nil-bit entries.